src/Overview/ProjectLane/BranchCard/BranchCard.js

```diff
@@ -4,73 +4,85 @@ import ProgressLane from './CardLanes/ProgressLane/ProgressLane.js';
 import UserLane from './CardLanes/UserLane/UserLane.js';
 
 class BranchCard extends Component {
-  constructor(props) {
-    super(props);
-    this.state = {branchName: this.props.branch.getName()};
-    this.branch = this.props.branch;
-  }
-  componentDidMount() {
-    if(this.branch.isIssueExtractable()) {
-      this.setState({issueNumber: this.branch.getIssueNumber(), issueDescription: this.branch.getIssueName()});
+    constructor(props) {
+        super(props);
+        this.state = {branchName: this.props.branch.getName(),
+                        branch: this.props.branch};
+    }
+    componentDidMount() {
+        this.extractInformation(this.props);
+    }
+
+    componentWillReceiveProps(nextProps) {
+        this.extractInformation(nextProps);
     }
-  }
 
-  getTitleForCard() {
-    if (this.state.issueNumber && this.state.issueDescription) {
-      return (
-        <span>
+    extractInformation(props) {
+        if (this.state.branchName !== props.branch.getName()) {
+            this.setState({branchName: props.branch.getName()});
+        }
+        if(props.branch.isIssueExtractable()) {
+            this.setState({issueNumber: props.branch.getIssueNumber(), issueDescription: props.branch.getIssueName()});
+        }
+        this.setState({branch: props.branch});
+    }
+
+    getTitleForCard() {
+        if (this.state.issueNumber && this.state.issueDescription) {
+            return (
+                <span>
           <p>{this.state.issueDescription}</p>
-          <a href={this.branch.getJiraLink()} className="white-text" target="_blank">
+          <a href={this.state.branch.getJiraLink()} className="white-text" target="_blank">
             #{this.state.issueNumber}
           </a>
         </span>
-      );
-    }
-    else {
-      return this.state.branchName;
+            );
+        }
+        else {
+            return this.state.branchName;
+        }
     }
-  }
 
-  getColorBasedOnLatestBuild() {
-    if(!this.branch.didTravisRun()) {
-      return "grey darken-2";
+    getColorBasedOnLatestBuild() {
+        if(!this.state.branch.didTravisRun()) {
+            return "grey darken-2";
```

Comment:
You are right, but I didn't want to do this in this refactoring branch :) But we should definitely consider some constants here.